### PR TITLE
V3.6.0 geometry compatibility

### PR DESCRIPTION
**Pull request by stanleyljl**
Re: #

### Changelog

* compatibility for geometry renderer

-------

### Continuous Integration

This pull request:

* [ ] needs automatic test cases check.

  > Manual trigger with `@cocos-robot run test cases` afterward.

-------

### Compatibility Check

This pull request:

* [ ] changes public API, and have ensured backward compatibility with [deprecated features](https://github.com/cocos/cocos-engine/blob/v3.5.0/docs/contribution/deprecated-features.md).
* [ ] affects platform compatibility, e.g. system version, browser version, platform sdk version, platform toolchain, language version, hardware compatibility etc.
* [ ] affects file structure of the build package or build configuration which requires user project upgrade.
* [ ] **introduces breaking changes**, please list all changes, affected features and the scope of violation.

### CODE REVIEW

File: cocos/core/pipeline/deprecated.ts [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+ THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ THE SOFTWARE.
+ */
+
+import { markAsWarning } from '../utils';
```
Comment (minggo) [inline cocos/core/pipeline/deprecated.ts:26]: I think there is a rule for deprecating file name. @pandamicro is there a documentation about it?

File: cocos/core/pipeline/deprecated.ts [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+import { RenderPipeline } from './render-pipeline'; 
+
+
+// deprecate RenderPipeline API
+markAsWarning(RenderPipeline.prototype, 'RenderPipeline.prototype', [
```
Comment (minggo) [inline cocos/core/pipeline/deprecated.ts:30]: Should add comment mention which version to deprecate it.

File: native/cocos/renderer/pipeline/RenderPipeline.h [hunk 3]
```diff
@@ -142,6 +147,7 @@ class CC_DLL RenderPipeline : public Asset {
     PipelineUBO *_pipelineUBO{nullptr};
     scene::Model *_profiler{nullptr};
     IntrusivePtr<PipelineSceneData> _pipelineSceneData;
+    GeometryRenderer *_geometryRenderer{nullptr};
```
Comment (minggo) [inline native/cocos/renderer/pipeline/RenderPipeline.h:150]: So, it is a weak reference?

File: native/cocos/renderer/pipeline/RenderPipeline.cpp [hunk 4]
```diff
@@ -124,6 +125,20 @@ void RenderPipeline::destroyQuadInputAssembler() {
...
+    // Query the first camera rendering to swapchain.
+    for (const auto *camera : cameras) {
+        if (camera && camera->getWindow() && camera->getWindow()->getSwapchain() ) {
+            _geometryRenderer = camera->getGeometryRenderer();
+            return;
+        }
+    }
+}
+
 bool RenderPipeline::destroy() {
```
Comment (minggo) [inline native/cocos/renderer/pipeline/RenderPipeline.cpp:142]: Should set `_geometryRenderer ` to `nullptr` here?

File: cocos/core/pipeline/deprecated.ts (hunk 1 above)
Comment (stanleyljl) [reply cocos/core/pipeline/deprecated.ts:26]: This is how panda taught me.

File: native/cocos/renderer/pipeline/RenderPipeline.h (hunk 3 above)
Comment (stanleyljl) [reply native/cocos/renderer/pipeline/RenderPipeline.h:150]: Yes, it's assigned from a raw pointer type

File: native/cocos/renderer/pipeline/RenderPipeline.cpp (hunk 4 above)
Comment (stanleyljl) [reply native/cocos/renderer/pipeline/RenderPipeline.cpp:142]: ok

**Review by minggo — APPROVED** [review]

**Comment by suqionglian** [comment]
@cocos-robot run test cases

**Review by star-e — APPROVED** [review]

**Comment by suqionglian** [comment]
@cocos-robot run test cases

**Pull request merged**